src/pipeline/activator/activator/planet/main.py

```diff
@@ -0,0 +1,275 @@
+    parser.add_argument("--s3-prefix", type=str, default=os.environ.get("S3_PREFIX", "planet-scene-cogs"))
+    parser.add_argument("--stac-api-uri", type=str, default=os.environ.get("STAC_API_URI", "http://franklin:9090"))
+    parser.add_argument("--temp-dir", type=str, default='/tmp')
+
+    parser.add_argument('--no-download', required=False, dest='download', action='store_false')
+    parser.set_defaults(download=True)
+
+    parser.add_argument('--no-upload', required=False, dest='upload', action='store_false')
+    parser.set_defaults(upload=True)
+
+    parser.add_argument('--no-update', required=False, dest='update', action='store_false')
+    parser.set_defaults(update=True)
+
+    return parser
+
+
+def get_download_link(num_bands, planet_id, planet_api_uri, planet_api_key):
+    logger.info('Requesting status of imagery')
+    result = requests.get(planet_api_uri.format(
+        source_collection_id(num_bands), planet_id),
+                          auth=HTTPBasicAuth(planet_api_key, ''))
+    result = result.json()
+
+    logger.info('(Re)activating and getting download link')
+    links = result['analytic']['_links']
+    self_link = links['_self']
+    activation_link = links['activate']
+    result = requests.get(activation_link,
+                          auth=HTTPBasicAuth(planet_api_key, ''))
+    result = requests.get(self_link, auth=HTTPBasicAuth(planet_api_key, ''))
+    result = result.json()
+    is_active = result['status'] == 'active'
+
+    if not is_active:
+        raise Exception('Imagery inactive')
+    else:
+        download_link = result['location']
+
+    return download_link
+
+
+def download_from_planet(download_link,
+                         planet_id,
+                         planet_api_key,
+                         temp_dir,
+                         download):
+    filename_tiff = f'/{temp_dir}/{planet_id}.tiff'
+    logger.info(f'Downloading imagery to {filename_tiff}')
+    if download:
+        data = requests.get(download_link,
+                            auth=HTTPBasicAuth(planet_api_key, ''),
+                            stream=True)
+        with open(filename_tiff, 'wb') as f:
+            data.raw.decode_content = True
+            shutil.copyfileobj(data.raw, f)
+        os.system(f'gdaladdo {filename_tiff}')
+    return filename_tiff
+
+
+def upload_to_s3(filename_tiff, s3_bucket, s3_prefix, upload):
+    filename = filename_tiff.split('/')[-1]
+    s3_uri = f's3://{s3_bucket}/{s3_prefix}/{filename}'
+    logger.info(f'Uploading imagery to {s3_uri}')
+    if upload:
+        os.system(f'aws s3 cp {filename_tiff} {s3_uri}')
+    return s3_uri
+
+
+def generate_stac_item(filename_tiff, cog_collection, planet_id, s3_uri):
+    logger.info(f'Using gdalinfo to get metadata')
+    filename_json = filename_tiff.replace('.tiff', '.json')
+    os.system(f'gdalinfo -proj4 -json {filename_tiff} > {filename_json}')
```

Comment:
was not it easier to use gdal py bindings there? Or that is just easier since it is some sort of pattern you follow?